vg/vg.go

```diff
@@ -85,6 +88,49 @@ type CanvasSizer interface {
 	Size() (x, y Length)
 }
 
+// CanvasWriterTo wraps behavior required for writing
+// a Canvas representation to a concrete image.
+type CanvasWriterTo interface {
+	CanvasSizer
+	io.WriterTo
+}
+
+var (
+	formatLock sync.RWMutex
+	formats    = make(map[string]func(w, h Length) CanvasWriterTo)
```

Comment:
You are right wrt the semantics behind 'format'. I'm not yet convinced that we should be covering all cases with WriterTo, though I certainly could be. Currently I am focusing specifically towards storage formats.

>if only one could connect the Title of the plot.Plot to the canvas creation...

That's entirely feasible. Do you think a panic on "" is reasonable for OpenGL/X11, or is that not necessary?